pkg/promtail/targets/syslogtarget.go

```diff
@@ -0,0 +1,327 @@
+	syslogParsingErrors.Inc()
+}
+
+func (t *SyslogTarget) handleMessage(connLabels labels.Labels, msg syslog.Message) {
+	if msg.Message() == nil {
+		return
+	}
+
+	lb := labels.NewBuilder(connLabels)
+	if v := msg.SeverityLevel(); v != nil {
+		lb.Set("__syslog_message_severity", *v)
+	}
+	if v := msg.FacilityLevel(); v != nil {
+		lb.Set("__syslog_message_facility", *v)
+	}
+	if v := msg.Hostname(); v != nil {
+		lb.Set("__syslog_message_hostname", *v)
+	}
+	if v := msg.Appname(); v != nil {
+		lb.Set("__syslog_message_app_name", *v)
+	}
+	if v := msg.ProcID(); v != nil {
+		lb.Set("__syslog_message_proc_id", *v)
+	}
+	if v := msg.MsgID(); v != nil {
+		lb.Set("__syslog_message_msg_id", *v)
+	}
+
+	if t.config.LabelStructuredData && msg.StructuredData() != nil {
+		for id, params := range *msg.StructuredData() {
+			id = strings.Replace(id, "@", "_", -1)
+			for name, value := range params {
+				key := "__syslog_message_sd_" + id + "_" + name
+				lb.Set(key, value)
+			}
+		}
+	}
+
+	processed := relabel.Process(lb.Labels(), t.relabelConfig...)
+
+	filtered := make(model.LabelSet)
+	for _, lbl := range processed {
+		if strings.HasPrefix(lbl.Name, "__") {
+			continue
+		}
+		filtered[model.LabelName(lbl.Name)] = model.LabelValue(lbl.Value)
+	}
+
+	t.messages <- message{filtered, *msg.Message()}
+}
+
+func (t *SyslogTarget) messageSender() {
+	for msg := range t.messages {
+		if err := t.handler.Handle(msg.labels, time.Now(), msg.message); err != nil {
+			level.Error(t.logger).Log("msg", "error handling line", "error", err)
+		}
+		syslogEntries.Inc()
+	}
+}
+
+func (t *SyslogTarget) connectionLabels(c net.Conn) labels.Labels {
+	lb := labels.NewBuilder(nil)
+	for k, v := range t.config.Labels {
+		lb.Set(string(k), string(v))
+	}
+
+	ip := ipFromConn(c).String()
+	lb.Set("__syslog_connection_ip_address", ip)
+	lb.Set("__syslog_connection_hostname", lookupAddr(ip))
+
+	return lb.Labels()
+}
+
+func ipFromConn(c net.Conn) net.IP {
+	switch addr := c.RemoteAddr().(type) {
+	case *net.TCPAddr:
+		return addr.IP
+	}
+
+	return nil
+}
+
+func lookupAddr(addr string) string {
+	names, _ := net.LookupAddr(addr)
+	return strings.Join(names, ",")
+}
+
+// Type returns SyslogTargetType.
+func (t *SyslogTarget) Type() TargetType {
+	return SyslogTargetType
+}
+
+// Ready indicates whether or not the syslog target is ready to be read from.
+func (t *SyslogTarget) Ready() bool {
+	return true
+}
+
+// DiscoveredLabels returns the set of labels discovered by the syslog target, which
+// is always nil. Implements Target.
+func (t *SyslogTarget) DiscoveredLabels() model.LabelSet {
+	return nil
+}
+
+// Labels returns the set of labels that statically apply to all log entries
+// produced by the SyslogTarget.
+func (t *SyslogTarget) Labels() model.LabelSet {
+	return t.config.Labels
+}
+
+// Details returns target-specific details.
+func (t *SyslogTarget) Details() interface{} {
+	return map[string]string{}
+}
```

Comment:
I would love to get info useful for checking how does the target, I think we should create an issue, have you seen the /targets and /service-discovery page in Promtail?

Reply:
> I would love to get info useful for checking how does the target, I think we should create an issue, have you seen the /targets and /service-discovery page in Promtail?

Just took a look at those pages and indeed there is some useful information missing from the syslog target.

I think the most obvious candidates would be open connections in `/targets` and the listen address in `/service-discovery`.

Have to dig in a bit further to see how those pages look in different setups...